enterprise/core-edge/src/main/java/org/neo4j/coreedge/raft/log/segmented/SegmentFile.java

```diff
@@ -262,30 +224,44 @@ public long size()
         return fileSystem.getFileSize( file );
     }
 
-    public String getFilename()
+    String getFilename()
     {
         return file.getName();
     }
 
-    @Override
-    public void close() throws DisposedException
+    /**
+     * Called by the pruner when it wants to prune this segment. If there are no open
+     * readers or writers then the segment will be closed.
+     *
+     * @return True if the segment can be pruned at this time, false otherwise.
+     */
+    boolean tryClose()
     {
-        if ( closed )
+        if ( refCount.tryDispose() )
         {
-            throw new RuntimeException( format( CLOSED_ERROR_MESSAGE, file ) );
+            close();
+            return true;
         }
+        return false;
+    }
 
-        closed = true;
+    @Override
+    public void close()
+    {
         closeWriter();
-        readerPool.close();
+
+        if ( !refCount.tryDispose() )
+        {
+            log.warn( "Segment still referenced. Value: %d", refCount.get() );
```

Comment:
It isn't expected and would probably indicate some lifecycle issue or unclean shutdown of other component. Should be harmless in general, thus a warning and not an error.